core/domain/collection_services.py

```diff
@@ -932,18 +1091,32 @@ def publish_collection_and_update_user_profiles(committer, collection_id):
     Args:
         committer: UserActionsInfo. UserActionsInfo object for the committer.
         collection_id: str. ID of the collection to be published.
+
+    Raises:
+        Exception. If no collection summary model exists for the given id.
     """
-    rights_manager.publish_collection(committer, collection_id)
+    rights_manager.publish_collection(committer, collection_id)  # type: ignore[no-untyped-call]
     contribution_time_msec = utils.get_current_time_in_millisecs()
     collection_summary = get_collection_summary_by_id(collection_id)
+    if collection_summary is None:
+        raise Exception(
+            'No collection summary model exists for the give id: %s'
+            % collection_id
+        )
```

Comment:
Why not use `assert`?

Reply:
Before reaching this line of code we are already checking two times if the collection exists or not. I added a raise exception just for a more specific error message. If any reviewer has any concerns, I'm ok to revert it to assert!!

Reply:
I think it is fine too keep just the assert if we don't expect a `None` at this point in code.

Reply:
Done!

Reply:
> Here we are sure that collection_summary is never going to be none, because: This method is only used in collection_editor.py
> 
> https://github.com/oppia/oppia/blob/99eb242993656fb9defa534496fb0b147ddd8a7f/core/controllers/collection_editor.py#L233
> 
> 
> and there we are fetching the model with same `collection_id` keeping strict = true.
> https://github.com/oppia/oppia/blob/99eb242993656fb9defa534496fb0b147ddd8a7f/core/controllers/collection_editor.py#L225

Ohhh, Sorry for the wrong info!!

Basically here we checking for the collection model ( not for the collection summary model ). So due to the absence of  check for collection_summary model before this line, I think assert is applicable here.